src/device-api/v2.ts

```diff
@@ -488,7 +489,16 @@ export function createV2Api(router: Router, applications: ApplicationManager) {
 
 	router.get('/v2/cleanup-volumes', async (_req, res) => {
 		try {
-			await applications.volumes.removeOrphanedVolumes();
+			const targetState = await applications.getTargetApps();
+			const referencedVolumes: string[] = [];
+			_.each(targetState, app => {
+				_.each(app.volumes, vol => {
+					referencedVolumes.push(
+						Volume.generateDockerName(vol.appId, vol.name),
+					);
+				});
```

Comment:
nit: this might work too
```
referencedVolumens = _.flatMap(targetState, (app) => app.volumes)
  .map((vol) => Volume.generateDockerName(vol.appId, vol.name));
```

Reply:
It's a bit strange, the initial map gets a list of dictionaries, and because the names can be the same across applications, I wouldn't like to have lodash do the merging (if it even can).